app/src/main/java/org/literacyapp/receiver/ScreenOnReceiver.java

```diff
@@ -3,21 +3,43 @@
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
 
-import org.literacyapp.LiteracyApplication;
-import org.literacyapp.authentication.CameraViewActivity;
+import org.literacyapp.facerecognition.StudentImageCollectionActivity;
 
+import java.util.Calendar;
+
+/**
+ * If more than 30 minutes have passed since the last StudentImageCollection event, open iamge
+ * collection activity.
+ */
 public class ScreenOnReceiver extends BroadcastReceiver {
-    public ScreenOnReceiver() {
-    }
+
+    private static final String PREF_LAST_STUDENT_IMAGE_COLLECTION = "pref_last_student_image_collection";
 
     @Override
     public void onReceive(Context context, Intent intent) {
-        LiteracyApplication literacyApplication = (LiteracyApplication) context.getApplicationContext();
-        if (literacyApplication.TEST_MODE){
-            Intent cameraViewIntent = new Intent(context, CameraViewActivity.class);
-            cameraViewIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-            context.startActivity(cameraViewIntent);
+        Log.i(getClass().getName(), "onReceive");
+
+        Calendar calendarLastCollectionEvent = null;
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        long timeOfLastCollectionInMillis = sharedPreferences.getLong(PREF_LAST_STUDENT_IMAGE_COLLECTION, 0);
+        if (timeOfLastCollectionInMillis > 0) {
+            calendarLastCollectionEvent = Calendar.getInstance();
+            calendarLastCollectionEvent.setTimeInMillis(timeOfLastCollectionInMillis);
+            Log.i(getClass().getName(), "calendarLastCollectionEvent.getTime(): " + calendarLastCollectionEvent.getTime());
+        }
+        Calendar calendar30MinutesAgo = Calendar.getInstance();
+        calendar30MinutesAgo.add(Calendar.MINUTE, -30);
+        Log.i(getClass().getName(), "calendar30MinutesAgo.getTime(): " + calendar30MinutesAgo.getTime());
+        if ((calendarLastCollectionEvent == null) || (calendarLastCollectionEvent.before(calendar30MinutesAgo))) {
```

Comment:
What's missing here is to check if at least 30 minutes has passed with the screen turned off, or with the device turned off. So I'll add a way to storing the timestamp when the device/screen turns off.